$OpenBSD: patch-Source_cmComputeLinkInformation_cxx,v 1.10 2014/10/03 10:11:45 dcoppa Exp $

XXX Temporary workaround for a sporadic segfault which seems to
result from memory corruption. To be removed when CMake 3.1 will
be released.

--- Source/cmComputeLinkInformation.cxx.orig	Thu Sep 11 15:24:01 2014
+++ Source/cmComputeLinkInformation.cxx	Fri Sep 26 22:49:55 2014
@@ -255,7 +255,7 @@ cmComputeLinkInformation
     ->GetPropertyAsBool("FIND_LIBRARY_USE_OPENBSD_VERSIONING");
 
   // The configuration being linked.
-  this->Config = config;
+  this->Config = this->Makefile->GetSafeDefinition("CMAKE_BUILD_TYPE");
 
   // Allocate internals.
   this->OrderLinkerSearchPath =
@@ -685,10 +685,19 @@ void cmComputeLinkInformation::AddItem(std::string con
         }
       else
         {
-        // Use the full path given to the library file.
         this->Depends.push_back(item);
-        this->AddFullItem(item);
-        this->AddLibraryRuntimeInfo(item);
+        if(!this->OpenBSD)
+          {
+          // Use the full path given to the library file.
+          this->AddFullItem(item);
+          this->AddLibraryRuntimeInfo(item);
+          }
+        else
+          {
+          std::string file = cmSystemTools::GetFilenameName(item);
+          this->AddUserItem(file, false);
+          this->OrderLinkerSearchPath->AddLinkLibrary(item);
+          }
         }
       }
     else
@@ -1082,12 +1091,21 @@ void cmComputeLinkInformation::AddTargetItem(std::stri
     this->SharedLibrariesLinked.insert(target);
     }
 
-  // Handle case of an imported shared library with no soname.
-  if(this->NoSONameUsesPath &&
-     target->IsImportedSharedLibWithoutSOName(this->Config))
+  if(this->OpenBSD)
     {
-    this->AddSharedLibNoSOName(item);
-    return;
+    if(target->GetType() == cmTarget::SHARED_LIBRARY)
+      {
+      this->AddSharedLibNoSOName(item);
+      return;
+      }
+    } else {
+    // Handle case of an imported shared library with no soname.
+    if(this->NoSONameUsesPath &&
+       target->IsImportedSharedLibWithoutSOName(this->Config))
+      {
+      this->AddSharedLibNoSOName(item);
+      return;
+      }
     }
 
   // If this platform wants a flag before the full path, add it.
